File: qb-engineer-server/qb-engineer.core/Models/CreatePartRequestModel.cs
using QBEngineer.Core.Enums;

namespace QBEngineer.Core.Models;

public record CreatePartRequestModel(
    string Description,
    string? Revision,
    PartType PartType,
    string? Material,
    string? MoldToolRef,
    string? ExternalPartNumber);

File: qb-engineer-ui/src/app/features/parts/models/create-part-request.model.ts
import { PartType } from './part-type.type';

export interface CreatePartRequest {
  description: string;
  revision?: string;
  partType: PartType;
  material?: string;
  moldToolRef?: string;
  externalPartNumber?: string;
  toolingAssetId?: number;
  minStockThreshold?: number;
  reorderPoint?: number;
  reorderQuantity?: number;
  leadTimeDays?: number;
  safetyStockDays?: number;
}
